### PR TITLE
fix: log all drop_user calls with record details when possible

### LINKED ISSUE #650: Log drop_user calls and reason
We should know exactly when and why we dropped a user record, along with a snap shot of what the user-record looked like.

### PR DESCRIPTION
**Pull request by bbangert**
Closes #650

### CODE REVIEW

File: autopush/router/webpush.py [hunk 1]
```diff
@@ -64,16 +68,18 @@ def preflight_check(self, uaid_data, channel_id):
         """Verifies this routing call can be done successfully"""
         uaid = uaid_data["uaid"]
         if 'current_month' not in uaid_data:
-            self.log.info("Record missing 'current_month' {record}",
-                          record=repr(uaid_data))
+            self.log.info(format="Dropping User", code=102,
```
Comment (jrconlin) [inline autopush/router/webpush.py]: I like that we log the action, should we still log the cause? It's an info so it may help.

Comment (bbangert) [inline autopush/router/webpush.py]: I numbered every code uniquely so we could identify where in the code-base we executed it from. The reason in a few cases is the same or similar, so knowing the origin of the call seemed more important than potentially the same reason in multiple places.

Comment (jrconlin) [inline autopush/router/webpush.py]: Ah, missed that the errno is different for each call.

**Comment by jrconlin** [comment]
r+

File: autopush/web/webpush.py [hunk 2]
```diff
@@ -63,8 +66,11 @@ def _router_completed(self, response, uaid_data, warning=""):
                 # An empty router_data object indicates that the record should
                 # be deleted. There is no longer valid route information for
                 # this record.
+                self.log.info(format="Dropping User", code=100,
```
Comment (pjenvey) [inline autopush/web/webpush.py]: This code isn't unique. I guess it really doesn't matter because it's in a separate file, but FYI

**Comment by pjenvey** [comment]
r+

**Comment by kitcambridge** [comment]
LGTM!

**Pull request merged**